### PR TITLE
duck 4.7.5

### PR DESCRIPTION
**Pull request by dkocher**

### CODE REVIEW

File: Library/Formula/duck.rb [hunk 1]
```diff
@@ -16,11 +16,12 @@ class Duck < Formula
   depends_on :java => ["1.8+", :build]
   depends_on :xcode => :build
   depends_on "ant" => :build
+  depends_on "maven" => :build
 
   def install
     revision = version.to_s.rpartition(".").last
-    system "ant", "-Dbuild.compile.target=1.8", "-Drevision=#{revision}", "cli"
-    libexec.install Dir["build/duck.bundle/*"]
+    system "mvn", "-DskipTests", "-Drevision=#{revision}", "--projects", "cli/osx", "--also-make", "verify"
```
Comment (DomT4) [inline Library/Formula/duck.rb]: You'll need to add `ENV.java_cache` to the top of `def install` here.

**Comment by dkocher** [comment]
Looks like we have some dependencies not available publicly. We will get this fixed.

**Comment by DomT4** [comment]
Thanks! Shout when they've been published and I'll get CI to do this again.

**Comment by mikemcquaid** [comment]
@dkocher Do you test these formulae before you submit them?

**Comment by dkocher** [comment]
@mikemcquaid I feel your frustration why this submittal is so bumpy. The cause is that we changed our build process but have run it only in our continuous integration system where there were some additional dependencies installed in a private repository which should be public.

**Comment by mikemcquaid** [comment]
@dkocher Ok. It'd be good if your CI system could test the Homebrew formula before it's submitted. Thanks!

**Comment by dkocher** [comment]
We only run `brew audit --strict --online` in the CI to test the formulae. Is it possibly to test the full installation in a sandbox?

**Comment by bfontaine** [comment]
@dkocher You can use `brew install --sandbox`.

**Comment by dkocher** [comment]
@bfontaine Does not look like this works.
````
xcloud954:target Xcloud$ brew install --sandbox homebrew-cyberduck/duck.rb 
Error: duck-4.7.3.18396 already installed
To install this version, first `brew unlink duck`
````
I would expect it not to complain about the version already installed in `/usr/local`. I cannot find any documentation on the `--sandbox` option.

**Comment by bfontaine** [comment]
If you want to replicate the CI you can use `brew test-bot <formula>`.

**Comment by mikemcquaid** [comment]
Or just have a dedicated Homebrew installation you use only for testing by doing `git clone https://github.com/Homebrew/Homebrew` into an arbitrary directory and adding that `bin` into your `$PATH`.

**Comment by DomT4** [comment]
How's this going?

**Comment by mikemcquaid** [comment]
Closing in favour of a future PR that works.

**Pull request closed without merging**